bundle/terranova/deploy_mutator.go

```diff
@@ -0,0 +1,169 @@
+package terranova
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/terranova/db"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/jsonloader"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/databricks-sdk-go"
+)
+
+type terranovaDeployMutator struct{}
+
+func TerranovaDeploy() bundle.Mutator {
+	return &terranovaDeployMutator{}
+}
+
+func (m *terranovaDeployMutator) Name() string {
+	return "TerranovaDeploy"
+}
+
+func (m *terranovaDeployMutator) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+
+	client := b.WorkspaceClient()
+
+	cacheDir, err := b.CacheDir(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	b.ResourceDatabase, err = db.LoadOrCreate(cacheDir)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	countDeployed := 0
+
+	_, err = dyn.MapByPattern(
+		b.Config.Value(),
+		dyn.NewPattern(dyn.Key("resources"), dyn.AnyKey(), dyn.AnyKey()),
+		func(p dyn.Path, v dyn.Value) (dyn.Value, error) {
+			section := p[1].Key()
+			resourceName := p[2].Key()
+
+			spec, ok := specs[section]
+			if !ok {
+				log.Warnf(ctx, "Resource section not supported: %s", section)
+				return v, nil
+			}
+
+			err = deployResource(ctx, b.ResourceDatabase, client, spec, section, resourceName, v)
+			if err != nil {
+				diags = diags.Extend(diag.FromErr(err))
+			}
+			countDeployed += 1
+			return v, nil
+		},
+	)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+
+	// Not uploading at the moment, just logging to match the output.
+	if countDeployed > 0 {
+		cmdio.LogString(ctx, "Updating deployment state...")
+	}
+
+	return diags
+}
+
+func deployResource(ctx context.Context, db *db.JSONFileDB, client *databricks.WorkspaceClient, spec IResource, section, resourceName string, config dyn.Value) error {
```

Comment:
This function seems to be pretty long and difficult to follow, any way it can be split?

Reply:
TBH I don't see how it would be easier to follow if it was split.

Reply:
For example, splitting creating and updating flow into separate function can help